src/neuroglancer/point/base.ts

```diff
@@ -0,0 +1,74 @@
+/**
+ * @license
+ * Copyright 2016 Google Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {SliceViewChunkSource, SliceViewChunkSpecification, SliceViewChunkSpecificationBaseOptions, SliceViewSourceOptions} from 'neuroglancer/sliceview/base';
+import {getCombinedTransform} from 'neuroglancer/sliceview/base';
+import {ChunkLayout} from 'neuroglancer/sliceview/chunk_layout';
+import {partitionArray} from 'neuroglancer/util/array';
+import {approxEqual} from 'neuroglancer/util/compare';
+import {kAxes, kZeroVec, mat4, rectifyTransformMatrixIfAxisAligned, transformVectorByMat4, vec3} from 'neuroglancer/util/geom';
+import {SharedObject} from 'neuroglancer/worker_rpc';
+
+export interface RenderLayer { sources: PointChunkSource[][]|null; }
+
+export interface PointSourceOptions extends SliceViewSourceOptions {}
+
+export interface PointChunkSource extends SliceViewChunkSource { spec: PointChunkSpecification; }
+;
+
+export interface PointChunkSpecificationOptions extends SliceViewChunkSpecificationBaseOptions {
+  chunkDataSize: vec3;
+}
+
+/**
+ * Specifies a chunk layout and voxel size.
+ */
+export class PointChunkSpecification extends SliceViewChunkSpecification {
+  chunkBytes: number;
```

Comment:
It seems like this could just be part of the backend chunk class, rather than here in base, since it isn't needed in the frontend.  Additionally, I would guess that the number of bytes may vary between chunks (which would be another reason to make it a variable of the Chunk class, rather than the SliceViewChunkSpecification), rather than being fixed for all chunks for a given source.